src/Runtime/OMInstrument.inc

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===--------- OMTensor.inc - C/C++ Neutral OMTensor Implementation--------===//
+//
+// Copyright 2019-2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains implementations of OMTensor data structures
+// and helper functions.
+//
+//===----------------------------------------------------------------------===//
+
+#ifdef __cplusplus
+#include <cassert>
+#include <map>
+#include <numeric>
+#include <random>
+#include <string>
+#include <typeinfo>
+#include <vector>
+#else
+#include <assert.h>
+#endif
+
+#ifdef __APPLE__
+#include <stdlib.h>
+#else
+#include <malloc.h>
+#endif
+
+#include <stdio.h>
+#include <string.h>
+
+#include "onnx-mlir/Runtime/OMInstrument.h"
+
+#include <sys/time.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+
+static struct timeval globalTimeVal, initTimeVal;
+static bool timeEnabled = true;
+static bool virtualMemEnabled = true;
+static pid_t mypid;
+static char memCommand[200];
+
+void OMInstrumentInit() {
+  if (getenv("OMINSTRUMENTTIME")) {
+	  timeEnabled = !timeEnabled;
+  }
+  if (getenv("OMINSTRUMENTMEMORY")) {
+	  virtualMemEnabled = !virtualMemEnabled;
+  }
+
+  if (timeEnabled) {
+    gettimeofday(&globalTimeVal, NULL);
+    initTimeVal = globalTimeVal;
+  }
+
+  if (virtualMemEnabled) {
+     mypid = getpid();
+     sprintf(memCommand, "ps -o vsz --noheader -p %d", mypid);
+  }
+}
+
+void OMInstrumentPoint(int64_t id, int64_t tag) {
+  if (timeEnabled) {
+    struct timeval newTimeValue, result;
+    gettimeofday(&newTimeValue, NULL);
+    printf("ID=%s TAG=%ld ",(char *)&id, tag);
```

Comment:
see earlier comment about having an enum for the tag. Also, for the op, can we print something more informative than a number?

Reply:
The ID is changed to op name string. 
Enumerate is not used for tag because I am think tag may be a bit mask to provide more control on instrumentation.